actix/src/sec-5-arbiter.md

```diff
@@ -121,19 +121,23 @@ fn main() {
 
     // Define an execution flow using futures
     //
-    // `Addr::send` responds with a `Request`, which implements `Future`
-    // When awaited or mapped, it will resolve to a `Result<usize, MailboxError>`
+    // Start by sending a `Value(6, 7)` to our `SumActor`.
+    // `Addr::send` responds with a `Request`, which implements `Future`.
+    // When awaited or mapped, it will resolve to a `Result<usize, MailboxError>`.
     let execution = sum_addr
         .send(Value(6, 7))
-        // turn from `Future<usize, MailboxError>` to `Future<usize, ()>` with `.map_err`
+        // `.map_err` turns `Future<usize, MailboxError>` into `Future<usize, ()>`
+        //   and prints any `MailboxError`s we encounter
         .map_err(|e| {
             eprintln!("Encountered mailbox error: {:?}", e);
         })
-        // chain another computation onto the future (assuming the send was successful)
-        // returning a future from `and_then` chains that computation.
-        // In this case, becasue the `Arbiter` is executing the futures, the result of
-        // `dis_addr.send` (also a `Response`) will be spawned and executed on the Arbiter in the
-        // same thread.
+        // Assuming the send was successful, chain another computation
+        //   onto the future. Returning a future from `and_then` chains
+        //   that computation to the end of the existing future.
+        //
+        // In this case, because the `Arbiter` is executing the futures,
+        //   the result of `dis_addr.send` (also a `Response`) will be
+        //   spawned and executed on the Arbiter in the same thread.
```

Comment:
I don't think we need to mention threads at all here, just that the second computation occurs only after the first is complete and successful.

Reply:
So one of my biggest pain points in learning the library has been knowing the threading model generally. I guess I overemphasized it a little in the comments here, but I've seen a lot of people wondering how to integrate tokio thread pools with the library which I think is a symptom of not understanding the threading model.

Reply:
I like the improved comments in the code *a lot*, I think that the mention of threads isn't needed given the huge amount of documentation about the threading model about 30 cm up the page :) better to draw attention here to the "A and_then B" pattern that is executed, which I think you have done well.

Reply:
In case of futures, you should focus on event loop/arbiter terminology, rather than just thread. Since `Arbiter` implies thread that runs async IO

Reply:
@DoumanAsh So my thought process with that is basically this:

Out of context, this future could theoretically be running the chained computations on different threads, right? As I understand it, the future describes the action, not the runtime, so you could have a runtime that runs the chained parts in a thread-agnostic way. However, in this case, because the `Arbiter` is going to run it later (not at this section), we know the `.and_then` would be run on the same thread. However, I don't think it's really relevant in this case. Maybe it would be more important to showcase an example of it being relevant, if ever. I think `ActorFuture` basically covers this because it's very clear when the `Actor` and `Context` are mutable references, so I'll just take this part out.